mypy/checkexpr.py

```diff
@@ -1840,89 +1828,234 @@ def check_op_local(self, method: str, base_type: Type, arg: Expression,
                                context, arg_messages=local_errors,
                                callable_name=callable_name, object_type=object_type)
 
+    def check_op_reversible(self,
+                            op_name: str,
+                            left_type: Type,
+                            left_expr: Expression,
+                            right_type: Type,
+                            right_expr: Expression,
+                            context: Context) -> Tuple[Type, Type]:
+        # Note: this kludge exists mostly to maintain compatibility with
+        # existing error messages. Apparently, if the left-hand-side is a
+        # union and we have a type mismatch, we print out a special,
+        # abbreviated error message. (See messages.unsupported_operand_types).
+        unions_present = isinstance(left_type, UnionType)
+
+        def make_local_errors() -> MessageBuilder:
+            """Creates a new MessageBuilder object."""
+            local_errors = self.msg.clean_copy()
+            local_errors.disable_count = 0
+            if unions_present:
+                local_errors.disable_type_names += 1
+            return local_errors
+
+        def lookup_operator(op_name: str, base_type: Type) -> Optional[Type]:
+            """Looks up the given operator and returns the corresponding type,
+            if it exists."""
+            if not self.has_member(base_type, op_name):
+                return None
```

Comment:
I would skip this step at all. It caused several bugs in the past, i think we can just use `analyze_member_access `.

Reply:
Unfortunately, removing this check ended up breaking a few test cases -- namely testIntPow, which tests our pow plugin, and testMetaclassOpAccessAny, which tests operators + metaclasses. ([Logs here, if you're curious](https://gist.github.com/Michael0x2a/1d5a5bcb7f55c940874808e3712165cb))

I didn't really want to dive into either method in this PR, so I left a TODO for now. (I can also file an issue/add info to an existing issue if you want.)